site/docs/latest/security/tls.md

```diff
@@ -23,139 +23,134 @@ title: Encryption and Authentication using TLS
 
-## Configure the broker for TLS
+### CLI tools
 
-To configure a Pulsar {% popover broker %} to use TLS authentication, you'll need to make some changes to the `broker.conf` configuration file, which is located in the `conf` directory of your [Pulsar installation](../../getting-started/LocalCluster).
+[Command-line tools](../../reference/CliTools) like [`pulsar-admin`](../../reference/CliTools#pulsar-admin), [`pulsar-perf`](../../reference/CliTools#pulsar-perf), and [`pulsar-client`](../../reference/CliTools#pulsar-client) use the `conf/client.conf` config file in a Pulsar installation.
 
-Add these values to the configuration file (substituting the appropriate certificate paths where necessary):
+You'll need to add the following parameters to that file to use TLS authentication with Pulsar's CLI tools:
 
 ```properties
-# Enable TLS and point the broker to the right certs
-tlsEnabled=true
-tlsCertificateFilePath=/path/to/broker-cert.pem
-tlsKeyFilePath=/path/to/broker-key.pem
-tlsTrustCertsFilePath=/path/to/cacert.pem
-
-# Enable the TLS auth provider
-authenticationEnabled=true
-authorizationEnabled=true
-authenticationProviders=org.apache.pulsar.broker.authentication.AuthenticationProviderTls
+webServiceUrl=https://broker.example.com:8443/
+brokerServiceUrl=pulsar+ssl://broker.example.com:6651/
+useTls=true
+tlsAllowInsecureConnection=false
+tlsTrustCertsFilePath=/path/to/ca.cert.pem
+authPlugin=org.apache.pulsar.client.impl.auth.AuthenticationTls
+authParams=tlsCertFile:/path/to/my-role.cert.pem,tlsKeyFile:/path/to/my-role.key-pk8.pem
 ```
 
-{% include message.html id="broker_conf_doc" %}
+### Java client
 
-## Configure the discovery service
+```java
+import org.apache.pulsar.client.api.PulsarClient;
 
-The {% popover discovery %} service used by Pulsar brokers needs to redirect all HTTPS requests, which means that it needs to be trusted by the client as well. Add this configuration in `conf/discovery.conf` in your Pulsar installation:
-
-```properties
-tlsEnabled=true
-tlsCertificateFilePath=/path/to/broker-cert.pem
-tlsKeyFilePath=/path/to/broker-key.pem
+PulsarClient client = PulsarClient.builder()
+    .serviceUrl("pulsar+ssl://broker.example.com:6651/")
+    .enableTls(true)
+    .tlsTrustCertsFilePath("/path/to/ca.cert.pem")
+    .authentication("org.apache.pulsar.client.impl.auth.AuthenticationTls",
+                    "tlsCertFile:/path/to/my-role.cert.pem,tlsKeyFile:/path/to/my-role.key-pk8.pem")
+    .build();
 ```
 
-## Configure clients
+### Python client
 
-For more information on Pulsar client authentication using TLS, see the following language-specific docs:
+```python
+from pulsar import Client, Authentication
 
-* [Java client](../../clients/Java)
-* [C++ client](../../clients/Cpp)
+auth = Authentication("tls",
+                      "tlsCertFile:/path/to/my-role.cert.pem,tlsKeyFile:/path/to/my-role.key-pk8.pem")
```

Comment:
This should be `AuthenticationTLS('/path/to/my-role.cert.pem', '/path/to/my-role.key-pk8.pem')

Reply:
This works too (since the fix for auth with functions).

Reply:
Sure, but I think that by preference we should encourage the `AuthenticationTLS()` since it won't require to learn the string format encoding of the parameters.

The Python also has docs that are show in IDEs and at http://pulsar.apache.org/api/python/

Reply:
Changed.